torch/autograd/variable.py

```diff
@@ -63,8 +63,14 @@ def __getattr__(self, name):
         raise AttributeError(name)
 
     def __getitem__(self, key):
-        if isinstance(key, Variable) and type(key.data).__name__ == 'ByteTensor':
-            return MaskedSelect.apply(self, key)
+        if torch.is_tensor(key):
+            key = Variable(key)  # auto-wrap tensors
+        if isinstance(key, Variable):
+            if type(key.data).__name__ == 'ByteTensor':
+                return MaskedSelect.apply(self, key)
+            elif type(key.data).__name__ == 'LongTensor':
+                return IndexSelect.apply(self, 0, key)
```

Comment:
So, the reason that this is specific to Byte and Long is that these are the only "indexing" tensor types?